src/frontend/src/btc/utils/btc-transactions.utils.ts

```diff
@@ -0,0 +1,29 @@
+import type { BtcTransactionUi } from '$btc/types/btc';
+import type { BtcAddress } from '$lib/types/address';
+import type { BitcoinOutput, BitcoinTransaction } from '$lib/types/blockchain';
+import { nonNullish } from '@dfinity/utils';
+
+export const mapBtcTransaction = ({
+	transaction: { inputs, block_index, out, hash, time },
+	btcAddress
+}: {
+	transaction: BitcoinTransaction;
+	btcAddress: BtcAddress;
+}): BtcTransactionUi => {
+	const isTypeSend = inputs.some(({ prev_out }) => prev_out.addr === btcAddress);
+	const blockIndexAvailable = nonNullish(block_index);
+	const output: BitcoinOutput | undefined = out.find(({ addr }) =>
+		isTypeSend ? addr !== btcAddress : addr === btcAddress
+	);
+
+	return {
+		id: hash,
+		timestamp: time,
+		value: BigInt(output?.value ?? 0),
+		status: blockIndexAvailable ? 'confirmed' : 'pending',
+		blockNumber: block_index ?? undefined,
```

Comment:
You default to `undefined` because it can be `null` or `blockNumber: block_index` would be the same?

Reply:
Yes, the Blockchain API returns either `null` or `number`, and for the UI type I want to have `number | undefined`.